src/server/common.cc

```diff
@@ -349,10 +349,6 @@ GenericError ExecutionState::GetError() const {
   return err_;
 }
 
-const Cancellation* ExecutionState::GetCancellation() const {
-  return this;
-}
-
 void ExecutionState::ReportCancelError() {
```

Comment:
This function is still being used (in replication and in other places).  The issue here is that after this function is called `IsError()` is `true`. This is a bug and the correct values should be `false` and `IsCancelled true`. 

You do not update the state == `Cancelled` here but this is also not needed. See my top comment regarding the 3 states you introduced.